airflow/providers/yandex/secrets/secrets_manager.py

```diff
@@ -0,0 +1,280 @@
+
+    When ``{"config_prefix": "airflow/config"}`` is set, if a secret is defined with
+    the path ``airflow/config/sql_alchemy_conn``, the config with key ``sql_alchemy_conn`` would be
+    accessible.
+
+    When the prefix is empty, keys will use the Lockbox Secrets without any prefix.
+
+    .. code-block:: ini
+
+        [secrets]
+        backend = airflow.providers.yandex.secrets.secrets_manager.SecretsManagerBackend
+        backend_kwargs = {"yc_connection_id": "<connection_ID>", "folder_id": "<folder_ID>"}
+
+    You need to specify credentials or id of yandexcloud connection to connect to Yandex Lockbox with.
+    Credentials will be used with this priority:
+
+    * OAuth Token
+    * Service Account JSON file
+    * Service Account JSON
+    * Yandex Cloud Connection
+
+    If no credentials specified, default connection id will be used.
+
+    Also, you need to specify the Yandex Cloud folder ID to search for Yandex Lockbox secrets in.
+
+    :param yc_oauth_token: Specifies the user account OAuth token to connect to Yandex Lockbox with.
+        Looks like ``y3_xxxxx``.
+    :param yc_sa_key_json: Specifies the service account auth JSON.
+        Looks like ``{"id": "...", "service_account_id": "...", "private_key": "..."}``.
+    :param yc_sa_key_json_path: Specifies the service account auth JSON file path.
+        Looks like ``/home/airflow/authorized_key.json``.
+        File content looks like ``{"id": "...", "service_account_id": "...", "private_key": "..."}``.
+    :param yc_connection_id: Specifies the connection ID to connect to Yandex Lockbox with.
+        Default: "yandexcloud_default"
+    :param folder_id: Specifies the folder ID to search for Yandex Lockbox secrets in.
+        If set to None (null in JSON), requests will use the connection folder_id if specified.
+    :param connections_prefix: Specifies the prefix of the secret to read to get Connections.
+        If set to None (null in JSON), requests for connections will not be sent to Yandex Lockbox.
+        Default: "airflow/connections"
+    :param variables_prefix: Specifies the prefix of the secret to read to get Variables.
+        If set to None (null in JSON), requests for variables will not be sent to Yandex Lockbox.
+        Default: "airflow/variables"
+    :param config_prefix: Specifies the prefix of the secret to read to get Configurations.
+        If set to None (null in JSON), requests for variables will not be sent to Yandex Lockbox.
+        Default: "airflow/config"
+    :param sep: Specifies the separator used to concatenate secret_prefix and secret_id.
+        Default: "/"
+    :param endpoint: Specifies an API endpoint.
+        Leave blank to use default.
+    """
+
+    def __init__(
+        self,
+        yc_oauth_token: str | None = None,
+        yc_sa_key_json: str | None = None,
+        yc_sa_key_json_path: str | None = None,
+        yc_connection_id: str | None = None,
```

Comment:
I would rather avoid to use any Airflow Connections into the Secrets Backend.

1. **Chicken and Egg problem**: Very easy to get into the situation that Backend not yet initialised
2. **Security/Maintainability**: Authenticated UI users who has access to the Connections (Metadata) might change creds and make Deployment Managers surprised. See: [Airflow Security Model](https://airflow.apache.org/docs/apache-airflow/stable/security/security_model.html)

Reply:
@potiuk @eladkal
I want to validate my concerns, I don’t want to force to rewrite half of the logic if it just my paranoia 🙄

Reply:
My thoughts here is:
1. This problem is addressed in this PR
2. I guess Deployment Manager can just choose to not use Connection in SecretManager configuration if that connection metadata is not trustworthy enough.

On the other hand, ability to not duplicate creds between SecretsManager and Hook looks profitable for simple installations.

Reply:
> This problem is addressed in this PR

That might be not such as straight forward. Secrets Manager has particular issue not only directly with connection but with other components and it is very difficult to detect in regular unit tests because during unit tests everything already initialised, however in regular usage it is not.

Something what we have in the past with other Secrets Backends:
- Try to obtain Connection to the DB from the secrets backend was raised circular import error: https://github.com/apache/airflow/pull/26784/files
- Try to obtain Fernet key from the secrets backend was raised circular import error.

This past issues not directly related to this PR, but still worthwhile to check it with this provider, I guess one of this might produce some interesting errors, or not.

>I guess Deployment Manager can just choose to not use Connection in SecretManager configuration if that connection metadata is not trustworthy enough.

Yeah, I've checked that couple other components also use conn_id, e.g. remote logging mechanism. So maybe this one not so critical than I thought initially

> On the other hand, ability to not duplicate creds between SecretsManager and Hook looks profitable for simple installations.

This is good idea to have some wrapper class / sets of function to convert provided parameters to the valid parameters for the Hook and SecretsManager, the same approach use into the Amazon Provider and I guess into the Google provider. My point here more about whether or not is a good idea to use Connection ID in Secrets Manager.

Reply:
I think 2. Is not really problematic. The user who can edit connections is highly priviledged already and can access (read/write) any connection credentials. In this case though it gives a little different capabilities - you could redirect all secret queries to a different place for example. Looking at Confidentiality, Integrity and Availability:

* I think it does not impact Confidentiality (because such change will not "leak" any Airflow secrets to users who have no access to them - more than what they already have).   
* It has a little potential to Impact Integrity - but no more than for example individually changing connections to point to different places for example and breaking integrity of the workflows - Connection UI Editing user already has those capabilities
* It has a potentnial to impact Availability - but no more than what the Connection UI Editing user already has by individually modifying connections.

So - I do not think it elevates privilledges of such user.